### PR TITLE
Add a CI check for do-not-merge label

### PR DESCRIPTION
**Pull request by mpalmi**
This PR adds a new CI check for the do-not-merge label. The check just fails if the label is present.

### CODE REVIEW

File: .github/workflows/do-not-merge-checker.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    # Runs on PRs to main and release branches
+    branches:
+      - main
+      - release/**
+
+jobs:
+  # checks that a do-not-merge label is not present for a PR
+  do-not-merge-check:
+    # If there is a `do-not-merge` label, or this comes from a fork (community contributor) we ignore this check
+    if: ${{ (github.repository_owner == 'hashicorp' && contains(github.event.pull_request.labels.*.name, 'do-not-merge')) }}
```
Comment (VioletHynes) [inline .github/workflows/do-not-merge-checker.yml:16]: Have we 100% validated that this works? In the past, new GHAs failing on community branches has been a persistent issue, so I want to make sure that we're not introducing a new issue here

**Review by VioletHynes — APPROVED** [review]
Looks great! Thanks for this :D

File: .github/workflows/do-not-merge-checker.yml (hunk 1 above)
Comment (VioletHynes) [reply .github/workflows/do-not-merge-checker.yml:16]: Oh, that's a fair point. In the past, we've stumbled for stuff like community submitted PRs don't have the permissions to run a GHA that adds a label. If community PRs have permissions to read labels, then I guess I don't see why not, yeah. There's some stuff that shouldn't run, like the milestone check (which essentially never applies to community PRs), but if this one runs without failing, I see no harm in it!

Comment (mpalmi) [reply .github/workflows/do-not-merge-checker.yml]: Thank you!

**Pull request merged**